node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -96,19 +97,18 @@ case class DataMessageHandler(
             }
           }
           newChainApi <- chainApi.processFilter(filter)
-        } yield {
-
-          Try {
-            val blockFilter =
-              BlockFilter.fromBytes(filter.filterBytes, filter.blockHash)
-            callbacks.onCompactFilterReceived.foreach(
-              _.apply(filter.blockHash, blockFilter))
-          } match {
+          blockFilterT = Try(
```

Comment:
I think it could be rewritten without matching a `Try[]`:
```suggestion
          blockFilter <- Future(BlockFilter.fromBytes(filter.filterBytes, filter.blockHash))
          _ <- callbacks.onCompactFilterReceived.foldLeft(FutureUtil.unit)(
                (acc, callback) =>
                  acc.flatMap(_ => callback(filter.blockHash, blockFilter)))
```